clients/drcachesim/tests/cache_replacement_policy_unit_test.cpp

```diff
@@ -105,9 +172,128 @@ unit_test_cache_lru_four_way()
     cache_lru_test.access_and_check_lru(ADDRESS_E, 2); // A E c D
 }
 
+void
+unit_test_cache_fifo_four_way()
+{
+    cache_policy_test_t<cache_fifo_t> cache_fifo_test(/*associativity=*/4,
+                                                      /*line_size=*/32,
+                                                      /*total_size=*/256);
+    cache_fifo_test.initialize_cache();
+
+    const addr_t ADDRESS_A = 0;
+    const addr_t ADDRESS_B = 64;
+    const addr_t ADDRESS_C = 128;
+    const addr_t ADDRESS_D = 256;
+    const addr_t ADDRESS_E = 320;
+    const addr_t ADDRESS_F = 384;
+    const addr_t ADDRESS_G = 448;
+    const addr_t ADDRESS_H = 512;
+
+    assert(cache_fifo_test.get_block_index(ADDRESS_A) ==
+           cache_fifo_test.get_block_index(ADDRESS_B));
+    assert(cache_fifo_test.get_block_index(ADDRESS_B) ==
+           cache_fifo_test.get_block_index(ADDRESS_C));
+    assert(cache_fifo_test.get_block_index(ADDRESS_C) ==
+           cache_fifo_test.get_block_index(ADDRESS_D));
+    assert(cache_fifo_test.get_block_index(ADDRESS_D) ==
+           cache_fifo_test.get_block_index(ADDRESS_E));
+    assert(cache_fifo_test.get_block_index(ADDRESS_E) ==
+           cache_fifo_test.get_block_index(ADDRESS_F));
+    assert(cache_fifo_test.get_block_index(ADDRESS_F) ==
+           cache_fifo_test.get_block_index(ADDRESS_G));
+    assert(cache_fifo_test.get_block_index(ADDRESS_G) ==
```

Comment:
How about a block_indices_are_identical() helper like the tags_are_different() to simplify this code (and in the other test functions).

Reply:
Done.